src/core/absorption_chiller.jl

```diff
@@ -28,61 +28,217 @@
+    installed_cost_per_ton::Union{Float64, Nothing} = nothing
+    om_cost_per_ton::Union{Float64, Nothing} = nothing
+
+
+    #Optional
+    min_ton::Float64 = 0.0,
+    max_ton::Float64 = 0.0,
+    cop_thermal::Union{Float64, Nothing} = nothing,
+    cop_electric::Float64 = 14.1,
+    om_cost_per_ton::Union{Float64, Nothing} = nothing,
+    macrs_option_years::Float64 = 0,
+    macrs_bonus_fraction::Float64 = 0
 ```
+
+!!! note "Required inputs"
+    To model AbsorptionChiller, you must provide at least one of the following: (i) `hot_water_or_steam` from $(heat_transfer_mediums), (ii) 
+    (ii), `chp_prime_mover` from $(chp_prime_movers),or (iii) all of the "custom inputs" defined below.
+    If prime_mover is provided, any missing value from the "custom inputs" will be populated from data/absorption_chiller/defaults.json, 
+    based on the `hot_water_or_steam` or `prime_mover`. boiler_type is "steam" if `prime_mover` is "combustion_turbine" 
+    and is "hot_water" for all other `prime_mover` types.
+
+    `fuel_cost_per_mmbtu` is always required
 """
-struct AbsorptionChiller <: AbstractThermalTech
-    min_ton::Real
-    max_ton::Real
-    cop_thermal::Real
-    cop_electric::Real
-    installed_cost_us_dollars_per_ton::Real
-    om_cost_us_dollars_per_ton::Real
-    macrs_option_years::Real
-    macrs_bonus_fraction::Real
-    min_kw::Real
-    max_kw::Real
-    installed_cost_per_kw::Real
-    om_cost_per_kw::Real
-    function AbsorptionChiller(;
-        min_ton::Real = 0.0,
-        max_ton::Real = 0.0,
-        cop_thermal::Real,
-        cop_electric::Real = 14.1,
-        installed_cost_per_ton::Real,
-        om_cost_per_ton::Real,
-        macrs_option_years::Real = 0,
-        macrs_bonus_fraction::Real = 0,
-        )
+Base.@kwdef mutable struct AbsorptionChiller <: AbstractThermalTech
+    hot_water_or_steam::Union{String, Nothing} = nothing
+    installed_cost_per_ton::Union{Float64, Nothing} = nothing
+    min_ton::Float64 = 0.0
+    max_ton::Float64 = 0.0
+    cop_thermal::Union{Float64, Nothing} = nothing
+    cop_electric::Float64 = 14.1
+    om_cost_per_ton::Union{Float64, Nothing} = nothing
+    macrs_option_years::Float64 = 0
+    macrs_bonus_fraction::Float64 = 0
+    min_kw::Float64 = NaN
+    max_kw::Float64 = NaN
+    installed_cost_per_kw::Float64 = NaN
+    om_cost_per_kw::Float64 = NaN
+end
+
+function AbsorptionChiller(d::Dict;
+        chp_prime_mover::Union{String, Nothing} = nothing,
+        existing_boiler::Union{ExistingBoiler, Nothing} = nothing
+    )
+
+    # convert Vector{Any} from JSON dictionary to Vector{Float64}
+    for (k, v) in d
+        if typeof(v) <: AbstractVector{Any}
+            d[k] = convert(Vector{Float64}, v)  # JSON.parsefile makes things Vector{Any}
+        end
+    end
+
+    absorp_chl = AbsorptionChiller(; dictkeys_tosymbols(d)...)
+
+    #check for 0.0 max size, return nothing if so
+    if absorp_chl.max_ton == 0.0
```

Comment:
Let's not require max_ton, and instead default to a big number, to be consistent with other techs which just require the tech name to be in the inputs.

Reply:
I've addressed this in 298eac373f9feec60c770487aa4089131721aece and left a TODO to employ the new const BIG_NUMBER elsewhere, since it's used for a few technologies right now.